api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -34,6 +34,7 @@ message Filter {
 
   // The name of the filter to instantiate. The name must match a
   // :ref:`supported filter <config_network_filters>`.
+  // [#protodoc-extension: Filter]
```

Comment:
Network filter?

Reply:
I think I would recommend using the extension category we use in the code for registration? E.g.

https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/include/envoy/server/filter_config.h#L128

@phlax this the category that is printed as part of this loop here:

https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/source/server/server.cc#L325-L328

Essentially I think we need to go from category -> name/type_url which we implicitly have in the registered factories structure but we will need to either get out of a C++ utility or more optimally encode in bazel somehow.

Reply:
(I realize that the extension categories are Envoy specific, but IMO that is fine for now until we sort out how to do per-client documentation.)

Reply:
so, trying to figure out the existing categories, because ultimately its these that need to match

im seeing that as part of protodoc an `extensions.db` json file is created, and checking the keys of this these do appear to be the categories that are marked in the protos

eg:

``` console
$ git grep "\[\#extension: envoy.access_loggers.http_grpc\]" api/
api/envoy/config/accesslog/v2/als.proto:// [#extension: envoy.access_loggers.http_grpc]
api/envoy/extensions/access_loggers/grpc/v3/als.proto:// [#extension: envoy.access_loggers.http_grpc]
api/envoy/extensions/access_loggers/grpc/v4alpha/als.proto:// [#extension: envoy.access_loggers.http_grpc]
```

what that doesnt give me afaict is the list of extensions for each extension

im wondering whether i can do this with some kind of (existing) bazel query/command to resolve these extensions, or whether as the above comment seems to suggest these need to be parsed from factories in the source code (or i guess a running binary)

Reply:
@htucch any input on ^^ would be very helpful - i can start trying to hack protodoc to parse the extensions if this is required, but would like to confirm it is before trying to do so

Reply:
The protos aren't marking categories, they are using the legacy extension names that we previously had prior to type URLs and `typed_config` becoming the standard way to indicate extension by type.

I think we should make a mandatory field in `envoy_cc_extension` that includes the extension category (which should match one of Envoy's already known types). This will allow `bazel query` analysis.

Reply:
this part i think is resolved by #14744